pkg/controllers/jobset_controller.go

```diff
@@ -578,8 +581,13 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 
 	// If enableDNSHostnames is set, update job spec to set subdomain as
 	// job name (a headless service with same name as job will be created later).
-	if dnsHostnamesEnabled(rjob) {
-		job.Spec.Template.Spec.Subdomain = GenSubdomain(js, rjob)
+	if dnsHostnamesEnabled(js) {
+		job.Spec.Template.Spec.Subdomain = js.Spec.Network.Subdomain
+	}
+
+	// If enableDNSHostnames is set, a Subdomain should be set
+	if dnsHostnamesEnabled(js) && js.Spec.Network.Subdomain == "" {
```

Comment:
since we have this check elsewhere, can you remove it from here pls?

Reply:
yep, will push after I can add the additional webhook test.